core/src/main/java/io/smallrye/openapi/runtime/io/securityrequirement/SecurityRequirementReader.java

```diff
@@ -82,19 +99,43 @@ private SecurityRequirementReader() {
      * @return SecurityRequirement model
      */
     public static SecurityRequirement readSecurityRequirement(AnnotationInstance annotationInstance) {
+        SecurityRequirement requirement = new SecurityRequirementImpl();
+        addSecurityRequirement(requirement, annotationInstance);
+        if (requirement.getSchemes().isEmpty()) {
```

Comment:
```suggestion
        // True when the annotation is missing the `name` property.
        if (requirement.getSchemes().isEmpty()) {
```

Reply:
Hmm, this is correct, but looking at it again, a `SecurityRequirement` annotation should never be missing the `name` property.

I didn't like that there was a code path where `addSecurityRequirement()` could actually not add the requirement and I definitely didn't want to end up adding an empty `SecurityRequirement` to the model in that case.

Can you shed any light on why we're checking for `null` when retrieving the name? I agree that we can add a comment here, but it should be clear that this is a path we expect to never be taken.

Reply:
I agree it seems like an impossibility that `name` would be missing (since it has no default value). The `null` check does not seem necessary in `addSecurityRequirement`.